src/compiler/transformers/classFields.ts

```diff
@@ -1315,16 +1356,26 @@ namespace ts {
                 ),
                 /*location*/ constructor ? constructor.body : undefined
             );
+
+            function visitBodyStatement(statement: Node) {
+                if (useDefineForClassFields && isParameterPropertyDeclaration(getOriginalNode(statement), constructor!)) {
+                    return undefined;
+                }
+
+                return visitor(statement);
+            }
         }
 
         /**
          * Generates assignment statements for property initializers.
          *
          * @param properties An array of property declarations to transform.
          * @param receiver The receiver on which each property should be assigned.
+         * @param insertionIndex How far after a found property to splice the expression statement, if not the end of the array.
          */
-        function addPropertyOrClassStaticBlockStatements(statements: Statement[], properties: readonly (PropertyDeclaration | ClassStaticBlockDeclaration)[], receiver: LeftHandSideExpression) {
-            for (const property of properties) {
+        function addPropertyOrClassStaticBlockStatements(statements: Statement[], properties: readonly (PropertyDeclaration | ClassStaticBlockDeclaration)[], receiver: LeftHandSideExpression, insertionIndex?: number) {

```

Comment:
I don't see a call to this that uses `insertionIndex`, is it needed?

Reply:
Aha, it used to be. No longer. Removing.